### PR TITLE
clean up of validate import, now uses a yaml file for maintainability 

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:

is part of https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/14282612/ . but needed cleanup before we could do anything with it 

### Is there any documentation that needs to be updated for this PR?
no

### CODE REVIEW

File: hack/validate-imports/validate-imports.go [hunk 1]
```diff
@@ -22,16 +29,69 @@ func validateUnderscoreImport(path string) error {
...
+type importValidator struct {
+	AllowedNames map[string][]string `json:"allowedImportNames"`
+}
+
+func initValidator() importValidator {
+	allowed := importValidator{}
+	err := yaml.Unmarshal(allowedNamesYaml, &allowed)
+	if err != nil {
+		log.Println(err)
+		return importValidator{}
```
Comment (ross-bryan) [inline hack/validate-imports/validate-imports.go:49]: why do we return the "empty", but initialized validator instead off err-ing out here? could this give us some false positive CI passing?

**Review by ross-bryan — DISMISSED** [review]
overall lgtm. left a comment/question.

File: hack/validate-imports/validate-imports.go (hunk 1 above)
Comment (facchettos) [reply hack/validate-imports/validate-imports.go]: I agree. It now exits with status code 1 :)

File: hack/validate-imports/allowed-import-names.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    - "maofake"
+  github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1:
+    - "mcv1"
+  github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned:
+    - "mcoclient"
+  github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/fake:
+    - "mcofake"
+  github.com/openshift/machine-config-operator/pkg/generated/clientset/versioned/typed/machineconfiguration.openshift.io/v1:
+    - "mcoclientv1"
+  k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake:
```
[the remark this answers is not in the document]
Comment (facchettos) [reply hack/validate-imports/allowed-import-names.yaml]: done

**Review by s-amann — DISMISSED** [review]

**Review by SudoBrendan — APPROVED** [review]
LGTM

**Review by petrkotas — APPROVED** [review]

**Review by s-amann — APPROVED** [review]

**Pull request merged**